classes/02class/aws-networking/README.md

```diff
@@ -0,0 +1,167 @@
+
+### In a traditional infrastructure, what establishes that I have a LAN (local area network)?
+
+A computer with a manual IP address configured constitutes in a LAN, although it won't be able to communicate with 
+
+- One computer connected directly to another by cable (very old method)
+- A computer sharing its wireless connection (WLAN)
+- Computers connected to a switch (switch is basically a controller that enables devices to talk to each other within ONE LAN - you can still have VLANS to segregate that ONE lan)
+
+
+### What if from my LAN, I need to connect to the wide area network (WAN), often called internet, or to a second LAN i.e. another office branch?
+
+For connecting multiple networks together, a router is needed. Either as a full hardware device or just an application working as a Router. This device will be in charge of knowing all existing paths, receive requests from a network and forward the packages to its destination.
+
+### Internet Protocol Address (IP)
+
+An IP is a numerical, human-readable way of assigning a computer to a network. For example ``. The protocol has two addressiong versions are available:
+
+- IPv4 as a 32-bit address. i.e.: 172.16.254.1
+- IPv6 as a 128-bits address. i.e.: 2001:db8:0:1234:0:567:8:1
+
+IPv6 was introduced a few years ago due to the growth of the internet and the depletion of available IPv4 addresses. IPv6 is out of context of this material. For reference: https://docs.aws.amazon.com/vpc/latest/userguide/get-started-ipv6.html
+
+Reference content for deep-dive information:
+
+- [Wikipedia IP_address](https://en.wikipedia.org/wiki/IP_address)
+- [AWS IP Addressing](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-ip-addressing.html)
+
+I highly recommend going deeply and trully understanding how networks work through the above reference links before moving on.
+
+
+## Basic networking
+
+What we need to understand to get started with networking in the cloud is how to do some basic network seggregation with subnets and ip addressing.
+
+Subnets are made to isolate traffic within a network and provide security on the data being transmitted. Moving forward our examples will be based on a four individual subnets scenario.
+
+It is a best practice to isolate networks that have databases or private systems running from workloads that can be accessible from the internet. The naming convention for these subnets are: 
+
+- `private` subnet
+  - Workloads like databases or identity management systems will run here
+  - Any service running in this subnet won't be accessible direct from public networks (like internet). To access applications 
+  - Services can go out to the internet through a nat gateway.
```

Comment:
We need to have the concept of NAT explained in order to introduce something like a NAT gateway.
Maybe put the public subnet first, and then the private..